프로그래머스/1/70128. 내적/내적.js

```diff
@@ -0,0 +1,7 @@
+function solution(a, b) {
+    let arr = []
+    for (let i = 0; i < a.length; i++) {
+        arr.push(a[i]*b[i])
+    }
+    return arr.reduce((acc, cur) => acc + cur, 0)
```

Comment:
```suggestion
return a.reduce((acc, cur, i) => acc + cur * b[i], 0);
```
현재 코드는 먼저 곱셈 결과를 담기 위한 새 배열 arr을 생성하고, 그 후에 reduce를 사용하여 합계를 계산한는군요 ..!! 
두 배열을 순회하면서 바로 합계를 구하는 방식으로 코드를 줄일 수 있지 않을까 생각이 들어요 !

Reply:
맞아요~ reduce에서 매개변수로 index를 사용하면 두 배열을 순회할 수 있답니다 :)